src/figuresClasses.ts

```diff
@@ -1,19 +1,94 @@
-export interface Figure {
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
 
+enum Color {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue',
 }
 
-export class Triangle {
+export interface Figure {
+  shape: Shape,
+  color: Color,
 
+  getArea: () => number,
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  readonly shape = Shape.Triangle;
 
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (
+      a >= b + c
+      || b >= a + c
+      || c >= a + b
+      || a <= 0
+      || b <= 0
+      || c <= 0
```

Comment:
Create a helper function that accepts any number of args and checks them. Reuse it with all shapes.